lib/Transformation.ts

```diff
@@ -1,20 +1,27 @@
 import * as RDF from 'rdf-js';
 import * as RDFString from 'rdf-string';
-import { Algebra as Alg } from 'sparqlalgebrajs';
+import {Algebra as Alg} from 'sparqlalgebrajs';
 
 import * as E from './expressions';
+import {AsyncExtensionApplication, SimpleApplication, SyncExtensionExpression, TermExpression} from './expressions';
 import * as C from './util/Consts';
+import {TypeURL as DT} from './util/Consts';
 import * as Err from './util/Errors';
 import * as P from './util/Parsing';
 
-import {
-  namedFunctions,
-  regularFunctions,
-  specialFunctions,
-} from './functions';
-import { TypeURL as DT } from './util/Consts';
+import {namedFunctions, regularFunctions, specialFunctions,} from './functions';
+import {AsyncExtensionFunction, AsyncExtensionFunctionCB} from './evaluators/AsyncEvaluator';
+import {SyncExtensionFunction, SyncExtensionFunctionCB} from './evaluators/SyncEvaluator';
+import {ExtensionFunctionError} from './util/Errors';
 
-export function transformAlgebra(expr: Alg.Expression): E.Expression {
+export type ExtensionFunctionCB = AsyncExtensionFunctionCB | SyncExtensionFunctionCB;
+
+export interface TransformAlgebraConfig {
```

Comment:
No need for this IMO (see comments below)

Reply:
I would like to keep `type ExtensionFunctionCallback = AsyncExtensionFunctionCallback | SyncExtensionFunctionCallback;` because the right part is half a line long and is quite present within the file. I think it serves it's purpose?

Reply:
Sure, we can revisit in the next iteration.